src/components/voyage/VoyageCrewSelect.tsx

```diff
@@ -620,43 +628,41 @@ const BestCrew = (props : {
 			}
 		}
 
-		return <div>
-				<br />
+		const primarySecondaryOutput = (sk: Skill) => {
+			let isPri = sk.skill === STTApi.playerData.character.voyage_descriptions[0].skills.primary_skill;
+			let isSec = sk.skill === STTApi.playerData.character.voyage_descriptions[0].skills.secondary_skill;
+			return (
+				<span>{isPri ? 'Primary' : isSec ? 'Secondary' : ''}</span>
+			)
+		};
+		const failOutput = (sk: Skill) => {
+			const sv = sk.core + (sk.range_min + sk.range_max) / 2;
+			const failSeconds = (sv * 0.045 + 34) * 60;
+			return (
+				<div>First hazard failure<br/>expected @ {formatTimeSeconds(failSeconds)}</div>
+			)
+		};
+		return <div className='voyage-crew-select'>
 				{props.state === 'inprogress' && <div className='ui medium centered text active inline loader'>Still calculating...</div>}
 				{selectSlotId !== undefined &&
 					<Dropdown
-					fluid
-					selection
-					options={buildOptions(selectSlotId)}
-					onChange={(e, { value }) => setChoice(selectSlotId, value as number)}
-					// value={}
+						fluid
+						selection
+						options={buildOptions(selectSlotId)}
+						onChange={(e, { value }) => setChoice(selectSlotId, value as number)}
+						// value={}
 					/>
 				}
 				<Card.Group>{crewSpans}</Card.Group>
-				<ul>
-					{ Object.keys(skill_aggregates).map(k => skill_aggregates[k]).map(skill => {
-						let isPri = skill.skill == STTApi.playerData.character.voyage_descriptions[0].skills.primary_skill;
-						let isSec = skill.skill == STTApi.playerData.character.voyage_descriptions[0].skills.secondary_skill;
-
-						const firstHazardFailureTime = (sv: number) => sv * 0.045 + 34;
-
-						const sv = skill.core + (skill.range_min + skill.range_max) / 2;
-						return (
-							<li key={skill.skill}>
-								<span className='quest-mastery'>
-									<img src={CONFIG.SPRITES['icon_' + skill.skill].url} height={18} /> &nbsp; {skill.core} ({skill.range_min}-
-									{skill.range_max})&nbsp;[{sv}]&nbsp;
-									{isPri ? ' (Pri) ' : ''}
-									{isSec ? ' (Sec) ' : ''}
-									<> - First hazard failure expected at: {formatTimeSeconds(firstHazardFailureTime(sv)*60)}</>
-								</span>
-							</li>
-						);
-					})}
-				</ul>
-
+				{
+					props.crewSelection && props.crewSelection.length &&
```

Comment:
Prints an errant "0" when no crew is selected. Also crewSelection will not be undefined, so this can instead be changed to
{props.crewSelection.length > 0 && <VoyageSkillsReadout ... />}